### PR TITLE
NMS-14234: Add howto document on Integrating OSGi modules in OpenNMS

### PR DESCRIPTION
**Pull request by cgorantla**
### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-14234

### CODE REVIEW

File: docs/modules/development/pages/osgi/osgi-and-opennms.adoc [hunk 1]
```diff
@@ -0,0 +1,135 @@
+[[OSGI-and-OpenNMS]]
+= OSGI Integration within OpenNMS
+
```
Comment (Bonrob2) [inline docs/modules/development/pages/osgi/osgi-and-opennms.adoc:3]: Could we add a short introduction here? Something like "OSGi integration lets you do x, y, z."
Do we need to define OSGi, or will readers know what it is?

File: docs/modules/development/pages/osgi/osgi-and-opennms.adoc [hunk 2]
```diff
@@ -0,0 +1,135 @@
+[[OSGI-and-OpenNMS]]
+= OSGI Integration within OpenNMS
```
Comment (Bonrob2) [inline docs/modules/development/pages/osgi/osgi-and-opennms.adoc:2]:
```suggestion
= OSGi Integration within OpenNMS
```

File: docs/modules/development/pages/osgi/osgi-and-opennms.adoc [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+----
+Host localhost
+    HostKeyAlgorithms +ssh-dss
+    StrictHostKeyChecking no
+UserKnownHostsFile=/dev/null
+----
+
+== Adding a new maven module
+
+This section describes how to integrate a new maven module into OpenNMS.
```
[the remark this answers is not in the document]
Comment (cgorantla) [reply docs/modules/development/pages/osgi/osgi-and-opennms.adoc]: yeah, it is the only Java code can be added. so they know.

File: docs/modules/development/pages/osgi/osgi-and-opennms.adoc [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+For OpenNMS, this needs to be added in `container/karaf/src/main/filtered-resources/etc/org.apache.karaf.features.cfg`.
+
+For Minion, this needs to be added in `features/minion/repository/src/main/resources/features.boot`.
+
+For Sentinel, this needs to be added in `features/container/sentinel/src/main/filtered-resources/etc/org.apache.karaf.features.cfg`
+
+=== Modules that need to be on Spring
+
+If a maven module should be loaded in Spring by default, that maven module should be included in `opennms-base-assembly/pom.xml`. Once you add a dependency here, it will be available in `target/opennms/lib`
```
[the remark this answers is not in the document]
Comment (cgorantla) [reply docs/modules/development/pages/osgi/osgi-and-opennms.adoc]: Slightly modified and updated!

File: docs/modules/development/pages/osgi/osgi-and-opennms.adoc (hunk 1 above)
Comment (cgorantla) [reply docs/modules/development/pages/osgi/osgi-and-opennms.adoc:3]: Should be covered by https://github.com/OpenNMS/opennms/pull/4614/commits/9b1d2d8444d1c0c5558fba0ac97087785d13e2cb

**Comment by cgorantla** [comment]
Thanks @Bonrob2 for the review. This is mostly targeted towards developers who are new to OpenNMS system.

**Review by RangerRick — DISMISSED** [review]
nice!

**Comment by cgorantla** [comment]
> Nice, thanks.
> 
> Need to put second sentence on a separate line, and update as follows:
> 
> This document explains how to integrate a Maven module (OSGi bundle) into the system.

Done!

**Review by Bonrob2 — APPROVED** [review]
LGTM!

**Pull request merged**